File: app/src/main/java/com/drdisagree/iconify/overlaymanager/NotificationManager.java
package com.drdisagree.iconify.overlaymanager;

import static com.drdisagree.iconify.common.Dynamic.TOTAL_NOTIFICATIONS;

import com.drdisagree.iconify.config.Prefs;
import com.drdisagree.iconify.utils.OverlayUtil;

import java.io.File;

public class NotificationManager {

    public static void enableOverlay(int n) {
        disable_others(n);
        enable_pack(n);

        if (!Prefs.getBoolean("IconifyComponentCR1.overlay") || !Prefs.getBoolean("IconifyComponentCR2.overlay")) {
            OverlayUtil.enableOverlay("IconifyComponentCR1.overlay");
            OverlayUtil.enableOverlay("IconifyComponentCR2.overlay");
        }
    }

    protected static void enable_pack(int n) {
        String path = "/system/product/overlay/IconifyComponentNFN" + n + ".apk";

        if (new File(path).exists()) {
            String overlay = "IconifyComponentNFN" + n + ".overlay";

            if (!Prefs.getBoolean(overlay))
                OverlayUtil.enableOverlay(overlay);
        }
    }

    public static void disable_pack(int n) {
        String path = "/system/product/overlay/IconifyComponentNFN" + n + ".apk";

        if (new File(path).exists()) {
            String overlay = "IconifyComponentNFN" + n + ".overlay";

            if (Prefs.getBoolean(overlay))
                OverlayUtil.disableOverlay(overlay);
        }
    }

    protected static void disable_others(int n) {
        for (int i = 1; i <= TOTAL_NOTIFICATIONS; i++) {
            if (i != n) {
                String path = "/system/product/overlay/IconifyComponentNFN" + i + ".apk";

                if (new File(path).exists()) {
                    String overlay = "IconifyComponentNFN" + i + ".overlay";

                    if (Prefs.getBoolean(overlay))
                        OverlayUtil.disableOverlay(overlay);
                }
            }
        }
    }
}

File: app/src/main/java/com/drdisagree/iconify/core/utils/overlay/managers/NotificationManager.kt
package com.drdisagree.iconify.core.utils.overlay.managers

import com.drdisagree.iconify.core.utils.SystemUtils
import com.drdisagree.iconify.core.utils.overlay.OverlayUtils.disableOverlay
import com.drdisagree.iconify.core.utils.overlay.OverlayUtils.enableOverlayExclusiveInCategory
import com.drdisagree.iconify.core.utils.overlay.OverlayUtils.enableOverlays
import com.drdisagree.iconify.core.utils.overlay.OverlayUtils.isOverlayEnabled
import com.drdisagree.iconify.data.common.Dynamic.TOTAL_NOTIFICATIONS
import com.drdisagree.iconify.data.common.Dynamic.isAtleastA14
import com.drdisagree.iconify.data.config.RPrefs
import com.drdisagree.iconify.data.keys.XposedKey

object NotificationManager {

    fun enableOverlay(n: Int) {
        disableOthers(n)
        enableOverlayExclusiveInCategory("IconifyComponentNFN$n.overlay")

        if (!isOverlayEnabled("IconifyComponentCR1.overlay") || !isOverlayEnabled("IconifyComponentCR2.overlay")) {
            enableOverlays("IconifyComponentCR1.overlay", "IconifyComponentCR2.overlay")
        }

        if (isAtleastA14) {
            var requireReload = false

            if (!RPrefs.getBoolean(XposedKey.FIX_NOTIFICATION_COLOR)) {
                RPrefs.putBoolean(XposedKey.FIX_NOTIFICATION_COLOR, true)
                requireReload = true
            }

            if (!RPrefs.getBoolean(XposedKey.FIX_NOTIFICATION_FOOTER_BUTTON_COLOR)) {
                RPrefs.putBoolean(XposedKey.FIX_NOTIFICATION_FOOTER_BUTTON_COLOR, true)
                requireReload = true
            }

            if (requireReload) {
                SystemUtils.restartSystemUI()
            }
        }
    }

    fun disableOverlay(n: Int) {
        disableOverlay("IconifyComponentNFN$n.overlay")

        if (isAtleastA14) {
            var requireReload = false

            if (RPrefs.getBoolean(XposedKey.FIX_NOTIFICATION_COLOR)) {
                RPrefs.putBoolean(XposedKey.FIX_NOTIFICATION_COLOR, false)
                requireReload = true
            }

            if (RPrefs.getBoolean(XposedKey.FIX_NOTIFICATION_FOOTER_BUTTON_COLOR)) {
                RPrefs.putBoolean(XposedKey.FIX_NOTIFICATION_FOOTER_BUTTON_COLOR, false)
                requireReload = true
            }

            if (requireReload) {
                SystemUtils.restartSystemUI()
            }
        }
    }

    private fun disableOthers(n: Int) {
        for (i in 1..TOTAL_NOTIFICATIONS) {
            RPrefs.putBoolean("IconifyComponentNFN$i.overlay", i == n)
        }
    }
}